python/extensions/python.bzl

```diff
@@ -74,28 +76,30 @@ def _python_impl(module_ctx):
         module_toolchain_versions = []
 
         for toolchain_attr in mod.tags.toolchain:
-            toolchain_version = toolchain_attr.python_version
-            toolchain_name = "python_" + toolchain_version.replace(".", "_")
+            toolchain_version = full_version(toolchain_attr.python_version)
```

Comment:
I think that supporting patch level specificity in the `python.toolchain` could be actually desirable for only one reason, but it would be a good reason enough to keep it - different Python toolchain versions come from different indygreg toolchain releases and some of them may have different packaging behaviour, like the last release contains the following changelog:
> CPython 3.9.16 -> 3.9.17
> CPython 3.10.11 -> 3.10.12
> CPython 3.11.3 -> 3.11.4
> setuptools 67.7.2 -> 68.0.0
> pip 23.1.2 -> 23.2.2
> musl 1.2.3 -> 1.2.4
> OpenSSL 1.1.1s -> 1.1.1u
> SQLite 3.41.2 -> 3.42.0
> libxzma 5.2.9 -> 5.2.12
> Release artifacts for Linux s390x are now published.
> _crypt extension module is now distributed as a standalone shared library (as opposed to being statically linked in libpython). This removes a hard dependency on libcrypt.so.1 for compatibility with modern Linux distributions. See https://github.com/indygreg/python-build-standalone/issues/173 for context. Affects Linux non-musl distributions.

So actually supporting pinning to the `X.Y.Z` could be an important feature to consider, because some users may want to stick with an older version than what is available in the `MINOR_MAPPING`.

What we should not support though is two toolchains of `X.Y.Z1` and `X.Y.Z2` at the same time being used where `X.Y` are the same.

I am inclined to leave the addition of the warning in a separate PR, if you are not against it.

Reply:
> different indygreg releases have different behavior

Ah, good point. I added that to the FR. 

> We should not support two toolchains with different patch versions

Oh? Why? That position would avoid several complications. Can you post your rationale to the FR issue?

re: warning in a separate PR: SGTM.